File: gdscript/src/main/java/gdscript/psi/impl/GdBitAndSignImpl.java
package gdscript.psi.impl;

import com.intellij.extapi.psi.ASTWrapperPsiElement;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import gdscript.psi.GdBitAndSign;
import gdscript.psi.GdVisitor;
import org.jetbrains.annotations.NotNull;

public class GdBitAndSignImpl extends ASTWrapperPsiElement implements GdBitAndSign {

  public GdBitAndSignImpl(@NotNull ASTNode node) {
    super(node);
  }

  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitBitAndSign(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

}

File: gdscript/src/main/kotlin/gdscript/psi/impl/GdBitAndSignImpl.kt
package gdscript.psi.impl

import com.intellij.extapi.psi.ASTWrapperPsiElement
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElementVisitor
import gdscript.psi.GdBitAndSign
import gdscript.psi.GdVisitor

class GdBitAndSignImpl(node: ASTNode) : ASTWrapperPsiElement(node), GdBitAndSign {
    fun accept(visitor: GdVisitor) {
        visitor.visitBitAndSign(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }
}
